File: src/main/java/ar/edu/um/tesoreria/rest/model/ArancelPorcentaje.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table(uniqueConstraints = {
		@UniqueConstraint(columnNames = { "aranceltipoId", "productoId" }) })
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class ArancelPorcentaje extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 2658422436465673615L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long arancelporcentajeId;

	private Integer aranceltipoId;
	private Integer productoId;
	private BigDecimal porcentaje = new BigDecimal(0);

}

File: src/main/java/ar/edu/um/tesoreria/rest/model/kotlin/ArancelPorcentaje.kt
package ar.edu.um.tesoreria.rest.model.kotlin

import ar.edu.um.tesoreria.rest.model.Auditable
import jakarta.persistence.Entity
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import jakarta.persistence.Id
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import java.math.BigDecimal

@Entity
@Table(
	uniqueConstraints = [
		UniqueConstraint(columnNames = ["aranceltipoId", "productoId"])
	]
)
data class ArancelPorcentaje(

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	var arancelporcentajeId: Long? = null,

	var aranceltipoId: Int? = null,
	var productoId: Int? = null,
	var porcentaje: BigDecimal = BigDecimal.ZERO

) : Auditable() 